proxy/router/src/cache.rs

```diff
@@ -104,9 +104,12 @@ impl<K: Hash + Eq, V, N: Now> Cache<K, V, N> {
             // Only whole seconds are used to determine whether a node should be retained.
             // This is intended to prevent the need for repetitive reservations when
             // entries are clustered in tight time ranges.
+            // NOTE: the `as_secs()` could be removed if we *know* `self.max_idle_age`
+            //       is always a whole number of seconds.
+            let max_idle_secs = self.max_idle_age.as_secs();
```

Comment:
note that we have to access `self.max_idle_age` outside of the closure passed to `retain()`; otherwise, it borrows `self` immutably whilst it is already borrowed mutably by `retain`. This also has the advantage of only calling `as_secs()` on it a single time per call.